test/standalone/stack_iterator/unwind.zig

```diff
@@ -2,26 +2,157 @@ const std = @import("std");
 const builtin = @import("builtin");
 const debug = std.debug;
 const testing = std.testing;
+const posix = std.posix;
+const native_arch = builtin.cpu.arch;
+const native_os = builtin.os.tag;
+const link_libc = builtin.link_libc;
 
-noinline fn frame3(expected: *[4]usize, unwound: *[4]usize) void {
-    expected[0] = @returnAddress();
+const max_stack_trace_depth = 32;
+
+const do_signal = switch (native_os) {
+    .wasi, .windows => false,
+    else => true,
+};
+
+var installed_signal_handler = false;
+var handled_signal = false;
+var captured_frames = false;
+
+const AddrArray = std.BoundedArray(usize, max_stack_trace_depth);
+
+// Global variables to capture different stack traces in.  Compared at the end of main() against "expected" array
+var signal_frames: AddrArray = undefined;
+var full_frames: AddrArray = undefined;
+var skip_frames: AddrArray = undefined;
+
+// StackIterator is the core of this test, but still worth executing the dumpCurrentStackTrace* functions
+// (These platforms don't fail on StackIterator, they just return empty traces.)
+const supports_stack_iterator =
+    (native_os != .windows) and // StackIterator is (currently?) POSIX/DWARF centered.
+    (native_arch != .wasm32) and
+    (native_arch != .wasm64); // wasm has no introspection
+
+// Getting the backtrace inside the signal handler (with the ucontext_t)
+// gets stuck in a loop on some systems:
+const expect_signal_frame_overflow =
+    (native_arch == .arm and link_libc) or // loops above main()
+    (native_arch == .aarch64); // non-deterministic, sometimes overflows, sometimes not
+
+// Getting the backtrace inside the signal handler (with the ucontext_t)
+// does not contain the expected content on some systems:
+const expect_signal_frame_useless =
+    (native_arch == .x86_64 and link_libc and builtin.abi.isGnu()) or // stuck on pthread_kill?
+    (native_arch == .x86_64 and link_libc and builtin.abi.isMusl() and builtin.omit_frame_pointer) or // immediately confused backtrace
+    (native_arch == .x86_64 and builtin.os.tag.isDarwin()) or // immediately confused backtrace
+    (native_arch == .aarch64 or native_arch == .aarch64_be) or // non-deterministic, sometimes overflows, sometimes confused
+    (native_arch == .riscv64 and link_libc) or // `ucontext_t` not defined yet
+    native_arch == .mips or // Missing ucontext_t. Most stack traces are empty ... (with or without libc)
+    native_arch == .mipsel or // same as .mips
+    native_arch == .mips64 or // same as .mips
+    native_arch == .mips64el or // same as .mips
+    native_arch == .powerpc64 or // dumpCurrent* useless, StackIterator empty, ctx-based trace empty (with or without libc)
+    native_arch == .powerpc64le; // same as .powerpc64
```

Comment:
nit:

```suggestion
    native_arch.isPowerPC64() or // dumpCurrent* useless, StackIterator empty, ctx-based trace empty (with or without libc)
```

Also what about `powerpc`?

Reply:
`powerpc` behaves the same as `powerpc64*` flavors, so upgraded this to a `native_arch.isPowerPC()` check.